bootloader/ubootenv/env_test.go

```diff
@@ -25,94 +25,113 @@ import (
 	"io/ioutil"
 	"os"
 	"path/filepath"
-	"strings"
 	"testing"
 
 	. "gopkg.in/check.v1"
 
 	"github.com/snapcore/snapd/bootloader/ubootenv"
+	"github.com/snapcore/snapd/testutil"
 )
 
 // Hook up check.v1 into the "go test" runner
 func Test(t *testing.T) { TestingT(t) }
 
-type uenvTestSuite struct {
+type uenvBaseSuite struct {
 	envFile string
 }
 
-var _ = Suite(&uenvTestSuite{})
+func (s *uenvBaseSuite) SetUpTest(c *C) {
+	s.envFile = filepath.Join(c.MkDir(), "uboot.env")
+}
+
+type uenvNativeTestSuite struct {
+	uenvBaseSuite
+}
+
+type uenvTextTestSuite struct {
+	uenvBaseSuite
+}
+
+var _ = Suite(&uenvNativeTestSuite{})
+var _ = Suite(&uenvTextTestSuite{})
 
-func (u *uenvTestSuite) SetUpTest(c *C) {
-	u.envFile = filepath.Join(c.MkDir(), "uboot.env")
+func (u *uenvNativeTestSuite) SetUpTest(c *C) {
+	u.uenvBaseSuite.SetUpTest(c)
 }
 
-func (u *uenvTestSuite) TestSetNoDuplicate(c *C) {
-	env, err := ubootenv.Create(u.envFile, 4096)
+func (u *uenvNativeTestSuite) TestSetNoDuplicate(c *C) {
+	env, err := ubootenv.Create(u.envFile, ubootenv.NativeFormat, 4096)
 	c.Assert(err, IsNil)
 	env.Set("foo", "bar")
 	env.Set("foo", "bar")
 	c.Assert(env.String(), Equals, "foo=bar\n")
 }
 
-func (u *uenvTestSuite) TestOpenEnv(c *C) {
-	env, err := ubootenv.Create(u.envFile, 4096)
+func (u *uenvNativeTestSuite) TestSanityNativeIsNotText(c *C) {
```

Comment:
I kind of understand the spirit of this tests but is a bit odd. let me think a bit...

Reply:
it's not strictly necessary, and is essentially covered by the already in depth tests for native format, but thought it was a nice sanity test